File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
       int res[] = new int[arr.length];
        for(int i =0 ;i < arr.length -1 ;i++){
            int mx = 0;
            for(int j= i+1; j < arr.length; j++){
                mx = Math.max(mx, arr[j]);
                }
            res[i] =  mx;
            }
        res[arr.length-1] = -1; 
        return res;
        }
        
    }

File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        res = [None]*len(arr)
        mx = -1
        for i in range(len(arr)-1, -1, -1):
            res[i] = mx
            mx = max(mx, arr[i])
        return res
        # res = []
        # for i in range(0, len(arr)-1):
        #     mx = 0
        #     for j in range(i +1, len(arr)):
        #         mx = max(mx, arr[j])
        #     res.append(mx)
        # res.append(-1)
        # return res